games-strategy/0ad/patches/0ad-0.0.22-alpha.patch

```diff
@@ -0,0 +1,627 @@
++$(OBJDIR)/lvm.o: ../../src/host/lua-5.1.4/src/lvm.c $(GCH) | prebuild
++	@echo $(notdir $<)
++	$(SILENT) $(CC) $(PCHINCLUDES) $(CFLAGS) -MF $(OBJDIR)/lvm.d -MT "$@" -o "$@" -c "$<"
++$(OBJDIR)/lcode.o: ../../src/host/lua-5.1.4/src/lcode.c $(GCH) | prebuild
++	@echo $(notdir $<)
++	$(SILENT) $(CC) $(PCHINCLUDES) $(CFLAGS) -MF $(OBJDIR)/lcode.d -MT "$@" -o "$@" -c "$<"
++$(OBJDIR)/lauxlib.o: ../../src/host/lua-5.1.4/src/lauxlib.c $(GCH) | prebuild
++	@echo $(notdir $<)
++	$(SILENT) $(CC) $(PCHINCLUDES) $(CFLAGS) -MF $(OBJDIR)/lauxlib.d -MT "$@" -o "$@" -c "$<"
++
++-include $(OBJECTS:%.o=%.d)
++-include $(GCH:%.h.gch=%.h.d)
+diff -ur original/0ad-0.0.22-alpha/build/premake/premake4/premake4.lua 0ad-0.0.22-alpha/build/premake/premake4/premake4.lua
+--- original/0ad-0.0.22-alpha/build/premake/premake4/premake4.lua	2012-08-19 18:49:02.033292288 +0000
++++ 0ad-0.0.22-alpha/build/premake/premake4/premake4.lua	2017-12-03 08:06:49.206569472 +0000
+@@ -65,12 +65,15 @@
+ 			buildoptions{ "-mmacosx-version-min=10.4" }
+ 			linkoptions { "-mmacosx-version-min=10.4" }
+ 
+-		configuration { "not windows", "not solaris", "not macosx" }
++		configuration { "not windows", "not solaris", "not 
++macosx", "not haiku" }
+ 			linkoptions { "-rdynamic" }
+ 			
+ 		configuration { "solaris" }
+ 			linkoptions { "-Wl,--export-dynamic" }
+ 
++		configuration { "haiku" }
++			linkoptions { "-Wl,--export-dynamic" }
+ 
+ 
+ --
+diff -ur original/0ad-0.0.22-alpha/build/premake/premake4.lua 0ad-0.0.22-alpha/build/premake/premake4.lua
+--- original/0ad-0.0.22-alpha/build/premake/premake4.lua	2017-07-06 17:29:49.012845056 +0000
++++ 0ad-0.0.22-alpha/build/premake/premake4.lua	2017-12-03 08:39:11.733216768 +0000
+@@ -832,6 +832,7 @@
+ 		windows = { "lib/sysdep/os/win", "lib/sysdep/os/win/wposix", "lib/sysdep/os/win/whrt" },
+ 		macosx = { "lib/sysdep/os/osx", "lib/sysdep/os/unix" },
+ 		bsd = { "lib/sysdep/os/bsd", "lib/sysdep/os/unix", "lib/sysdep/os/unix/x" },
++		haiku = { "lib/sysdep/os/unix" }
+ 	}
+ 	for i,v in pairs(sysdep_dirs[os.get()]) do
+ 		table.insert(source_dirs, v);
+diff -ur original/0ad-0.0.22-alpha/build/workspaces/update-workspaces.sh 0ad-0.0.22-alpha/build/workspaces/update-workspaces.sh
+--- original/0ad-0.0.22-alpha/build/workspaces/update-workspaces.sh	2016-09-02 16:18:57.049807360 +0000
++++ 0ad-0.0.22-alpha/build/workspaces/update-workspaces.sh	2017-12-03 09:13:04.544473088 +0000
+@@ -1,9 +1,10 @@
+ #!/bin/sh
+ 
+-if [ "$(id -u)" = "0" ]; then
+-   echo "Running as root will mess up file permissions. Aborting ..." 1>&2
+-   exit 1
+-fi
++#if [ "$(id -u)" = "0" ]; then
++#   echo "Running as root will mess up file permissions. Aborting ..." 
++#1>&2
++#   exit 1
++#fi
+ 
+ die()
+ {
+@@ -34,14 +35,19 @@
+     ;;
+ esac
+ 
++if [ "`uname -s`" = "Haiku" ]; then
++    #Haiku requires the g++-x86 compiler to compile C++11 applications.
++    MAKE="make CC=gcc-x86 CXX=g++-x86"
```

Comment:
Please do not hardcode _x86

Reply:
And it shouldn't be needed. Why is it detecting the wrong GCC? The recipe should be built as _x86, no?